components/coreapi/utxo.go

```diff
@@ -23,30 +29,82 @@ func getOutput(c echo.Context) (*ledgerstate.Output, error) {
 }
 
 func getOutputMetadata(c echo.Context) (*outputMetadataResponse, error) {
-	output, err := getOutput(c)
+	outputID, err := httpserver.ParseOutputIDParam(c, restapipkg.ParameterOutputID)
 	if err != nil {
 		return nil, err
 	}
-	outputID := output.OutputID()
-	slotCommitment := deps.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment()
-	spent, err := deps.Protocol.MainEngineInstance().Ledger.IsOutputSpent(outputID)
+
+	latestCommitment := deps.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment()
+
+	unspent, err := deps.Protocol.MainEngineInstance().Ledger.IsOutputUnspent(outputID)
 	if err != nil {
 		return nil, err
 	}
 
-	includedIndex := output.BlockID().Index()
-	commitment, err := deps.Protocol.MainEngineInstance().Storage.Permanent.Commitments().Load(includedIndex)
+	if unspent {
+		return newOutputMetadataResponse(outputID, latestCommitment)
+	}
+
+	return newSpentMetadataResponse(outputID, latestCommitment)
+}
+
+func newOutputMetadataResponse(outputID iotago.OutputID, latestCommitment *model.Commitment) (*outputMetadataResponse, error) {
+	output, err := deps.Protocol.MainEngineInstance().Ledger.Output(outputID.UTXOInput())
+	if err != nil {
+		return nil, err
+	}
+
+	includedSlotIndex := output.SlotIndexBooked()
+	if includedSlotIndex > latestCommitment.Index() {
+		return nil, ErrOutputNotCommitted
```

Comment:
I would not return an error here but maybe make the `IncludedCommitmentID` optional? This would signal the user that the output is not yet commited